src/libstd/sys/unix/thread.rs

```diff
@@ -125,16 +125,32 @@ impl Thread {
     }
 
     pub fn sleep(dur: Duration) {
+        fn get_secs(secs: &mut u64) -> libc::time_t {
+            if *secs > libc::time_t::max_value() as u64 {
+                *secs -= libc::time_t::max_value() as u64;
+                libc::time_t::max_value()
+            } else {
+                let copy = *secs;
+                *secs = 0;
+                copy as libc::time_t
+            }
+        }
+
+        let mut secs = dur.as_secs();
         let mut ts = libc::timespec {
-            tv_sec: dur.as_secs() as libc::time_t,
+            tv_sec: 0,
             tv_nsec: dur.subsec_nanos() as libc::c_long,
```

Comment:
Oh this'll want to be 0 so long as `secs > 0` because otherwise an extra number of nanoseconds is going to be slept each time. I think that to be robust you may just want to move creation of this `timespec` into the loop below to avoid having to re-initialize these fields and have a value leak from one iteration of the loop to the next.